src/GitHub.App/Services/GitClient.cs

```diff
@@ -424,16 +424,7 @@ public Task<bool> IsHeadPushed(IRepository repo)
 
             return Task.Factory.StartNew(() =>
             {
-                if (repo.Head.IsTracking)
-                {
-                    var trackedBranchTip = repo.Head.TrackedBranch.Tip;
-                    if (trackedBranchTip != null)
-                    {
-                        return repo.Head.Tip.Sha == trackedBranchTip.Sha;
-                    }
-                }
-
-                return false;
+                return repo.Head.TrackingDetails.AheadBy == 0;
```

Comment:
I'm guessing that `repo.Head.TrackingDetails` might be `null` if the remote branch has been deleted (e.g when the PR you're on has been closed). I believe that `repo.Head.TrackedBranch` will be `null` in this case.

Should we return true or false if the remote beanch has been deleted?